### DISCUSSION TITLE
Support external buffers

### DISCUSSION THREAD

**Discussion by xiyaowong** [vscode-neovim/vscode-neovim#2149] [Ideas]
External buffer is a concept in vscode-neovim. Buffers that are not corresponding to vscode textEditor are considered external buffers.

The common needs include terminal buffers and buffers controlled by Neovim plugins (such as UI for plugin managers, undotree, etc.).

Currently, using our synchronization logic to synchronize the content of these buffers, achieving performance, accuracy, and consistent styling is challenging.

Since these buffers typically represent plugin UIs, managing and rendering them should be achieved through implementing a UI component.

A feasible solution is to use custom editor, webview, etc. to implement a UI that manages and draws such buffers.

**Comment by justinmk** [comment]
> use custom editor, webview, etc. to implement a UI that manages and draws such buffers.

Will native vscode features such as auto-completion, keymaps, and other plugins, work in such a buffer?
